mulog-kinesis/src/com/brunobonacci/mulog/publishers/aws_utils.clj

```diff
@@ -12,11 +12,11 @@
                                         :request {:StreamName stream-name
                                                   :Records    records}})
         failed-record-count (:FailedRecordCount rsp)]
-    (if (or (nil? failed-record-count)
-          (not (zero? failed-record-count)))
-      (println
-        (format "mu/log kinesis stream '%s' publisher failure; Reason: %s" stream-name rsp)))))
-
-(defn create-records
-  [k d]
-  {:PartitionKey k :Data d})
\ No newline at end of file
+    (cond
+      (nil? failed-record-count)        (println (format
```

Comment:
i'm unclear why if there are no failed record it is a publisher failure.
In other words, if all records have been sent successfully (no failed records)
shouldn't the request be considered as successful?

Reply:
This check is done for “no such stream” case. Not sure if it makes sense to retry.

Reply:
how can you understand that `(nil? failed-record-count)` means `no such stream`???
surely there is a better way to check this condition

Reply:
Agree, `(nil? failed-record-count)` check can look as a counterintuitive solution but from what I understood `cognitect.aws` doesn't throw exceptions and always return something.
For the “no such stream” case the response will look like 
```
  {:__type "ResourceNotFoundException",
   :message "Stream KINESIS-LOCAL-STREAM-NAME under account 000000000000 not found.",
   :cognitect.anomalies/category :cognitect.anomalies/incorrect}
```
I assume that the lack of permissions will be wrapped in a similar response structure.

The better approach probably would be the check  for `__type` key presence.